cmake/modules/AddSwift.cmake

```diff
@@ -356,10 +393,11 @@ function(_add_variant_link_flags)
   endif()
 
   if(NOT "${SWIFT_${LFLAGS_SDK}_ICU_UC}" STREQUAL "")
-    list(APPEND result "-L${SWIFT_${sdk}_ICU_UC}")
+    link_directories("${SWIFT_${sdk}_ICU_UC}")
```

Comment:
Unfortunately, we have to resort to an `CMAKE_C_COMPILER_ID` check for linking directories and do it manually with `-L` or `/LIBPATH`